app/src/main/kotlin/com/simplemobiletools/gallery/pro/adapters/MediaAdapter.kt

```diff
@@ -505,15 +505,23 @@ class MediaAdapter(
         }
 
         val isRecycleBin = firstPath.startsWith(activity.recycleBinPath)
-        val baseString = if (config.useRecycleBin && !isRecycleBin) R.string.move_to_recycle_bin_confirmation else R.string.deletion_confirmation
+        val baseString =
+            if (config.useRecycleBin && !config.tempSkipRecycleBin && !isRecycleBin) R.string.move_to_recycle_bin_confirmation else R.string.deletion_confirmation
         val question = String.format(resources.getString(baseString), itemsAndSize)
-        DeleteWithRememberDialog(activity, question) {
-            config.tempSkipDeleteConfirmation = it
-            deleteFiles()
+
+        val callback = fun(remember: Boolean, skipRecycleBin: Boolean) {
```

Comment:
lets add the function itself as a callback to DeleteWithRememberDialog as used in multiple other places, this is hard to read

Reply:
like this?

```kotlin
val callback = object: DeleteWithRememberDialog.Callback {
    override fun onConfirm(remember: Boolean, skipRecycleBin: Boolean) {
        config.tempSkipDeleteConfirmation = remember
        ...
    }
}

DeleteWithRememberDialog(activity, question, config.useRecycleBin, callback)
```

Reply:
no, like
DeleteWithRememberDialog(activity, question, config.useRecycleBin) { rememer, skipRecycleBin ->
...
}
it is used in many dialogs, try finding one